tools/refguide_check.py

```diff
@@ -792,57 +792,43 @@ def _run_doctests(tests, full_name, verbose, doctest_warnings):
     runner = DTRunner(full_name, checker=Checker(), optionflags=flags,
                       verbose=verbose)
 
-    output = []
+    output = io.StringIO()
     success = True
-    def out(msg):
-        output.append(msg)
-
-    class MyStderr:
-        """
-        Redirect stderr to the current stdout
-        """
-        def write(self, msg):
-            if doctest_warnings:
-                sys.stdout.write(msg)
-            else:
-                out(msg)
-
-        # a flush method is required when a doctest uses multiprocessing
-        # multiprocessing/popen_fork.py flushes sys.stderr
-        def flush(self):
-            if doctest_warnings:
-                sys.stdout.flush()
+    tmp_stderr = sys.stdout if doctest_warnings else output
 
-    # Run tests, trying to restore global state afterward
-    with np.errstate(), np.printoptions(), redirect_stderr(MyStderr()):
+    @contextmanager
+    def TempCwd():
         cwd = os.getcwd()
         tmpdir = tempfile.mkdtemp()
-        try:
-            os.chdir(tmpdir)
-
-            # try to ensure random seed is NOT reproducible
-            np.random.seed(None)
-
-            ns = {}
-            for t in tests:
-                # We broke the tests up into chunks to try to avoid PSEUDOCODE
-                # This has the unfortunate side effect of restarting the global
-                # namespace for each test chunk, so variables will be "lost" after
-                # a chunk. Chain the globals to avoid this
-                t.globs.update(ns)
-                t.filename = short_path(t.filename, cwd)
-                # Process our options
-                if any([SKIPBLOCK in ex.options for ex in t.examples]):
-                    continue
-                fails, successes = runner.run(t, out=out, clear_globs=False)
-                if fails > 0:
-                    success = False
-                ns = t.globs
-        finally:
-            os.chdir(cwd)
-            shutil.rmtree(tmpdir)
+        os.chdir(tmpdir)
+        yield tmpdir
```

Comment:
You still need the `try` / `finally` around this `yield`

Reply:
(and the chair above, as it was before)

Reply:
I don't understand "the chair above, as it was before"

Reply:
Chdir is not in my mobile dictionary...